### ISSUE TITLE
[OCR] Build platform-owned annotation and training-feedback pipeline

### ISSUE THREAD

**Issue by Teakowa** [OWBastion/owbastion.com#102]
## Background

The platform already owns private screenshot submissions, OCR orchestration, player/private evidence reads, maintainer review, and persisted OCR evidence. OCRKit owns recognition and model lifecycle.

The missing layer is a durable annotation workflow that turns production OCR evidence into reviewed training feedback without making production screenshots automatically become training data.

Relying on the local OCRKit Studio as the primary annotation source does not scale: one maintainer must manually review every candidate, while the screenshot owner can often cheaply confirm whether visible OCR fields are correct.

## Architecture decision

Keep one authoritative owner for each fact:

- `OWBastion/owbastion.com` owns player OCR feedback, reviewed annotations, annotation provenance, feedback-request policy, and immutable dataset snapshots.
- `OWBastion/ocrkit` owns recognition evidence, field-level confidence/warnings, ROI extraction, ML-ready dataset materialization, train/holdout preparation, training, evaluation, model artifacts, release, and rollback.
- A player correction is an **annotation proposal**, never ground truth by itself.
- A production screenshot/evidence record is not automatically a training sample.
- Annotation feedback must not change the current Submission decision, challenge match, mastery outcome, or title Grant. The business-review path and ML-improvement path may share evidence but remain independently audited workflows.
- Players must not be asked to confirm every successful OCR result. Feedback requests should be targeted by uncertainty/information value, with a passive correction path always available.

## Feedback-request policy

Treat player feedback as an active-learning signal rather than a mandatory extra submission step.

The platform should derive a server-side feedback prompt from OCR quality, field-level confidence/status/warnings, field criticality, and optional sampling policy. Portal components render that decision; they must not contain scattered hard-coded confidence thresholds.

Expected behavior:

- **No prompt by default:** when all relevant player-visible fields are sufficiently reliable, show the recognition result normally and do not require confirmation.
- **Targeted confirmation:** when one or a small number of safe fields are uncertain/conflicting, ask the player to confirm or correct only those fields.
- **Grouped confirmation:** when several safe fields are uncertain but the screenshot remains usable, present a compact grouped check rather than forcing re-entry of every field.
- **Passive correction:** even when no prompt is generated, keep a low-friction “recognition is wrong” entry so confident-but-wrong cases can still be reported.
- **Calibration spot checks:** allow a low, configurable sample of otherwise high-confidence results to request confirmation so false-confident errors and confidence calibration can be measured. Do not hard-code a rollout percentage in this parent issue.
- **Severe quality failures:** unsupported/cropped/unusable evidence should follow the existing resubmission/manual-review policy rather than being converted into a player annotation task.

Do not reduce the policy to one average/composite confidence number. A critical low-confidence field must remain actionable even when other fields are highly confident. Field-specific policy/criticality is allowed; exact initial thresholds should be centralized and evidence-driven rather than embedded in Portal UI.

The player response must never be required to continue an otherwise valid high-confidence submission flow unless an existing business rule independently requires player action.

## Target flow

```text
private submission + OCR evidence
→ platform feedback policy
  ├→ no prompt / passive correction only
  ├→ targeted or grouped player confirmation/correction
  └→ severe quality failure → existing resubmission/review path
→ annotation proposal when feedback is submitted
→ maintainer accepts / edits+accepts / rejects
→ reviewed annotation
→ immutable dataset snapshot
→ OCRKit imports snapshot
→ ROI preparation + source-level train/holdout split
→ train → evaluate → release
```

## Child work

- [ ] #103 Player OCR-result confirmation and correction feedback
- [ ] #104 Maintainer annotation review and provenance workflow
- [ ] #105 Immutable reviewed dataset snapshots and OCRKit consumption contract

Dependency order:

`#103 → #104 → #105 → OWBastion/ocrkit#5 → OWBastion/ocrkit#6`

`OWBastion/ocrkit#3` may proceed in parallel. `OWBastion/ocrkit#4` is an optional later experiment and must not block the main path.

## Platform responsibilities

The platform should retain enough immutable provenance to explain every reviewed annotation later, including:

- source Submission/evidence identity;
- OCR model and layout version recorded for that recognition result;
- field/ROI identity where available;
- original OCR value and safe normalized/display value where applicable;
- the feedback-policy reason/category that exposed the prompt, when a prompt was generated;
- player proposal when present;
- final reviewed transcription/value;
- review actor and time;
- lifecycle state;
- dataset snapshot membership.

Do not expose internal OCR response payloads, numeric confidence thresholds, risk signals, private object URLs, other players' evidence, or annotation moderation internals to ordinary players.

## Cross-repository dependencies

- `OWBastion/ocrkit#3` remains an independent deterministic terminology-normalization improvement and may proceed in parallel.
- `OWBastion/ocrkit#4` is an experiment that should wait for representative reviewed annotations and should not block this pipeline.
- `OWBastion/ocrkit#5` consumes finalized platform dataset snapshots.
- `OWBastion/ocrkit#6` narrows the local Studio after the new import path is usable.
- OCRKit must not require direct platform database access. Dataset consumption must use an explicit private contract/export boundary.

## Security and privacy constraints

- Player screenshots remain private evidence.
- Dataset access is maintainer/service-only and must not expose signed URLs or broad R2 credentials.
- Player feedback may only target the authenticated player's own Submission evidence.
- Annotation writes are idempotent/audited as appropriate.
- Dataset snapshots are immutable once finalized; later corrections belong to a later snapshot rather than silently mutating model provenance.
- Preserve deletion/retention obligations for source evidence and document how a snapshot behaves if source evidence later becomes unavailable.
- Confidence, warnings, and prompt-reason signals exposed to the Portal must be reduced to the minimum presentation contract; do not expose raw internal scoring merely to drive UI.

## Acceptance criteria

- [ ] A high-confidence usable Submission does not require routine player confirmation.
- [ ] The platform can derive `none` / targeted / grouped feedback behavior from centralized OCR quality and field-level policy rather than Portal-side thresholds.
- [ ] A low-confidence or conflicting critical field can trigger confirmation even when other fields are highly confident.
- [ ] A player can give bounded feedback on prompted safe OCR fields from their own Submission without seeing raw/internal OCR payloads.
- [ ] A passive correction entry remains available for non-prompted results so confident-but-wrong cases can be reported.
- [ ] The policy supports configurable high-confidence spot checks for calibration without coupling them to business approval.
- [ ] Severe screenshot-quality failures continue through existing resubmission/review handling instead of being mislabeled as annotation tasks.
- [ ] Player feedback cannot directly approve a Submission, award/revoke progression, or become training truth.
- [ ] Maintainers can efficiently review proposals and produce an auditable reviewed annotation.
- [ ] Reviewed annotations preserve source OCR/model/layout provenance and do not overwrite original OCR evidence.
- [ ] An immutable dataset snapshot can be created from explicitly eligible reviewed annotations.
- [ ] OCRKit can consume the snapshot through a private, versioned contract without direct D1/R2 coupling to platform internals.
- [ ] Source-level train/holdout splitting and ML label generation remain OCRKit responsibilities.
- [ ] Privacy-negative tests prove ordinary players/public APIs cannot enumerate annotations, datasets, other users' screenshots, training metadata, or internal confidence policy.
- [ ] Relevant product/security/API/design documentation and the Feature Status Matrix are updated only to evidence-supported status.

## Non-goals

- Crowdsourcing arbitrary screenshots outside the authenticated Submission workflow.
- Letting player feedback directly alter grants, mastery runs, or review outcomes.
- Requiring players to confirm every OCR result.
- Showing raw numeric confidence or internal risk scoring to players.
- Hosting PaddleOCR training inside the web platform.
- Moving OCRKit model artifacts, checkpoints, or release channels into the platform.
- Automatically training or releasing a model whenever annotations change.
- Building reputation points, rewards, leaderboards, or gamification for annotators in the first phase.
- Adding an LLM adjudicator as a prerequisite.

**Comment by Teakowa** [comment]
All platform child slices are complete (#103, #104, #105), and the downstream OCRKit follow-ups #5 and #6 are also completed. Current `main` records player OCR feedback, maintainer annotation review, and reviewed dataset snapshots/OCRKit consumption contract as integration-tested. No remaining platform implementation scope is left in this parent; deployed verification can advance independently through the existing feature-status/runbook path.
